ansible/modules/hashivault/hashivault_oidc_auth_role.py

```diff
@@ -0,0 +1,266 @@
+    argspec['token_type'] = dict(required=False, type='str', default='default')
+    argspec['clock_skew_leeway'] = dict(required=False, type='int', default=0)
+    argspec['expiration_leeway'] = dict(required=False, type='int', default=0)
+    argspec['not_before_leeway'] = dict(required=False, type='int', default=0)
+
+    supports_check_mode=False
+
+    module = hashivault_init(argspec, supports_check_mode) 
+    result = hashivault_oidc_auth_role(module)
+    if result.get('failed'):
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+@hashiwrapper
+def hashivault_oidc_auth_role(module):
+    params = module.params
+    client = hashivault_auth_client(params)
+    mount_point = params.get('mount_point')
+    name = params.get('name')
+    state = params.get('state')
+    desired_state = dict()
+    current_state = dict()
+    exists = False
+    changed = False
+
+    token = params['token']
+    headers = {'X-Vault-Token': token}
+    url = params['url']
+    verify = params['verify']
+
+    # do not want a trailing slash in name and mount_point
+    if name[-1] == '/':
+        name = name.strip('/')
+    if mount_point[-1]:
+        mount_point = mount_point.strip('/')
+
+    desired_state['allowed_redirect_uris'] = params.get('allowed_redirect_uris')
+    desired_state['bound_audiences'] = params.get('bound_audiences')
+    desired_state['bound_claims'] = params.get('bound_claims')
+    desired_state['bound_subject'] = params.get('bound_subject')
+    desired_state['claim_mappings'] = params.get('claim_mappings')
+    desired_state['groups_claim'] = params.get('groups_claim')
+    desired_state['oidc_scopes'] = params.get('oidc_scopes')
+    desired_state['token_bound_cidrs'] = params.get('token_bound_cidrs')
+    desired_state['token_explicit_max_ttl'] = params.get('token_explicit_max_ttl')
+    desired_state['token_ttl'] = params.get('token_ttl')
+    desired_state['token_max_ttl'] = params.get('token_max_ttl')
+    desired_state['token_no_default_policy'] = params.get('token_no_default_policy')
+    desired_state['token_policies'] = params.get('token_policies')
+    desired_state['token_type'] = params.get('token_type')
+    desired_state['user_claim'] = params.get('user_claim')
+    desired_state['token_period'] = params.get('token_period')
+    desired_state['token_num_uses'] = params.get('token_num_uses')
+    desired_state['clock_skew_leeway'] = params.get('clock_skew_leeway')
+    desired_state['expiration_leeway'] = params.get('expiration_leeway')
+    desired_state['not_before_leeway'] = params.get('not_before_leeway')
+
+
+    # check if engine is enabled
+    try:
+      if (mount_point + "/") not in client.sys.list_auth_methods()['data'].keys():
+        return {'failed': True, 'msg': 'auth method is not enabled', 'rc': 1}
+    except:
+        if module.check_mode:
+            changed = True
+        else:
+            return {'failed': True, 'msg': 'auth mount is not enabled', 'rc': 1}
```

Comment:
can you update the output to say: "auth mount is not enabled or namespace is not created"